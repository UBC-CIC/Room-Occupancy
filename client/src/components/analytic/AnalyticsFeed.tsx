import * as React from "react";
import { Divider, Feed, Icon, Segment, SemanticICONS } from "semantic-ui-react";

interface IAnalyticsFeedProps {
  summaryHeader: string;
  summarySubheader: string;
  summaryDate: string;
  icon: SemanticICONS | undefined;
  metaHeader: string;
}

export function AnalyticsFeed(props: IAnalyticsFeedProps) {
  const { summaryHeader, summarySubheader, summaryDate, icon, metaHeader } =
    props;

  return (
    <Feed>
      <Feed.Event>
        <Feed.Content>
          <Feed.Summary>
            <Feed.User>{summaryHeader}</Feed.User>
            {" " + summarySubheader}
            <Feed.Date>{summaryDate}</Feed.Date>
          </Feed.Summary>
          <Feed.Meta>
            <Feed.Like>
              <Icon name={icon} />
              {metaHeader}
            </Feed.Like>
          </Feed.Meta>
        </Feed.Content>
      </Feed.Event>
      <Divider />
    </Feed>
  );
}
