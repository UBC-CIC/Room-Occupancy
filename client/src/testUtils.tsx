import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import {
  MemoryRouter,
  MemoryRouterProps,
  BrowserRouter,
} from "react-router-dom";

// Implement with creation of providers
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// Usage example:
// test("matches snapshot", () => {
//   // Replace 'Navbar' with your desired component and provide the props
//   testSnapshot(Navbar, { /* props for Navbar */ });
// });
const testComponentSnapshot = <P extends {}>(
  Component: React.ComponentType<P>,
  props: P,
  routerProps: MemoryRouterProps = {}
) => {
  const { asFragment } = render(
    <MemoryRouter {...routerProps}>
      <Component {...props} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
};

export * from "@testing-library/react";
export { customRender, testComponentSnapshot };
