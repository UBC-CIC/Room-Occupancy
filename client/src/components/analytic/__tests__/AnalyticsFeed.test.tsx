import React from "react";
import "@testing-library/jest-dom";
import { customRender } from "../../../testUtils";
import { AnalyticsFeed } from "../AnalyticsFeed";

describe("<AnalyticsFeed />", () => {
  const mockProps = {
    summaryHeader: "Test Header",
    summarySubheader: "Test Subheader",
    summaryDate: "2023-01-01",
    icon: undefined,
    metaHeader: "Test Meta",
  };

  test("should matchSnapshot", () => {
    const { asFragment } = customRender(
      <AnalyticsFeed {...mockProps}>Colin</AnalyticsFeed>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render props", () => {
    const { getByText } = customRender(
      <AnalyticsFeed {...mockProps}>Colin</AnalyticsFeed>
    );

    expect(getByText(mockProps.summaryHeader)).toBeInTheDocument();
    expect(getByText(mockProps.summarySubheader)).toBeInTheDocument();
    expect(getByText(mockProps.summaryDate)).toBeInTheDocument();
    expect(getByText(mockProps.metaHeader)).toBeInTheDocument();
  });
});
