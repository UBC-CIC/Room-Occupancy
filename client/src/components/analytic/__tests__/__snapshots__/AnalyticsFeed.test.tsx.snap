// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalyticsFeed /> should matchSnapshot 1`] = `
<DocumentFragment>
  <div
    class="ui feed"
  >
    <div
      class="event"
    >
      <div
        class="content"
      >
        <div
          class="summary"
        >
          <a
            class="user"
          >
            Test Header
          </a>
           Test Subheader
          <div
            class="date"
          >
            2023-01-01
          </div>
        </div>
        <div
          class="meta"
        >
          <a
            class="like"
          >
            <i
              aria-hidden="true"
              class="icon"
            />
            Test Meta
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
